.github/workflows/ci.yaml

```diff
@@ -16,75 +16,55 @@ on:
 jobs:
   lint:
     name: Lint
-    runs-on: ubuntu-latest
-    timeout-minutes: 5
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Install tox
-        # TODO: Consider replacing with custom image on self-hosted runner OR pinning version
-        run: python3 -m pip install tox
-      - name: Run linters
-        run: tox run -e lint
+    uses: canonical/data-platform-workflows/.github/workflows/lint.yaml@v13.0.0
 
   unit-test:
     name: Unit tests
     runs-on: ubuntu-latest
     timeout-minutes: 5
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
-      - name: Install tox
-        # TODO: Consider replacing with custom image on self-hosted runner OR pinning version
-        run: python3 -m pip install tox
+        uses: actions/checkout@v4
+      - name: Install tox & poetry
+        run: |
+          pipx install tox
+          pipx install poetry
       - name: Run tests
         run: tox run -e unit
 
   build:
-    name: Build charms
-    uses: canonical/data-platform-workflows/.github/workflows/build_charms_with_cache.yaml@v8.0.0
-    permissions:
-      actions: write # Needed to manage GitHub Actions cache
+    name: Build charm
+    uses: canonical/data-platform-workflows/.github/workflows/build_charm.yaml@v13.0.0
+    with:
+      cache: true
+
+  build-test:
+    name: Build test charm
+    uses: canonical/data-platform-workflows/.github/workflows/build_charm.yaml@v13.0.0
+    with:
+      cache: true
+      path-to-charm-directory: tests/integration/application-charm
 
   integration-test:
     strategy:
       fail-fast: false
       matrix:
-        tox-environments:
-          - integration
-    name: ${{ matrix.tox-environments }}
+        juju:
+          - agent: 2.9.46
+            libjuju: ^2
+          - agent: 3.1.7
+        cloud:
+          - lxd
+          - microk8s
```

Comment:
Runs the tests for juju 2 and 3, k8s and vm.

Reply:
**praise** Nice!!!!